### PR TITLE
BFD-2891: Configure maven release and deploy plugins

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[BFD-2891](https://jira.cms.gov/browse/BFD-2891)

**User Story or Bug Summary:**

As a BFD engineer I would like to have a centralized repository for official release images and build artifacts.

---

### What Does This PR Do?

This PR enables the use of maven release plugin to build and push tagged release artifacts and update pom versions using a standard process.  There are also minor changes to the bfd-data project code to make its behavior consistent with comments (release builds were broken without these changes).

A new maven project, `bfd-ops` has been added to compile the ops directory into an assembly so that it can also be tagged and pushed to CodeArtifact for use in deployments.

More information about [how to use the release plugin](https://github.com/CMSgov/beneficiary-fhir-data/wiki/2891-Maven-Release-Plugin) is available in the BFD Wiki.

To save space in CodeArtifact the modules that publish artifacts to CA are limited to:

* `bfd-db-migrator`
* `bfd-pipeline-app`
* `bfd-server-launcher`
* `bfd-server-war`
* `bfd-ops`

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/pom.xml [hunk 1]
```diff
@@ -67,6 +68,8 @@
             they are reproducible bit-by-bit no matter when/where they are generated.
             See https://maven.apache.org/guides/mini/guide-reproducible-builds.html -->
         <project.build.outputTimestamp>2022-01-01T00:00:00Z</project.build.outputTimestamp>
+        <bfd.data.fda.version>1.0-SNAPSHOT</bfd.data.fda.version>
+        <bfd.data.npi.version>1.0-SNAPSHOT</bfd.data.npi.version>
```
Comment (lsmitchell) [inline apps/pom.xml:72]: I suppose we will be fixing these versions not matching the format of the rest when we eventually give them a real version.

File: apps/pom.xml [hunk 2]
```diff
@@ -1204,6 +1213,23 @@
     </build>
 
     <profiles>
+        <profile>
+            <id>test-release</id>
```
Comment (lsmitchell) [inline apps/pom.xml:1217]: You mentioned we have to adjust the name of this; will we do this later or now?

**Review by lsmitchell — APPROVED** [review]

File: apps/pom.xml (hunk 1 above)
Comment (brianburton) [reply apps/pom.xml:72]: Hopefully we'll replace these with a pipeline soon.  If we ever do another deploy of these I agree we should change them to use the same numbering scheme as the rest of BFD.

**Review by mjburling — APPROVED** [review]
🥇

**Pull request merged**